examples/vision/gradient_centralization.py

```diff
@@ -0,0 +1,282 @@
+"""
+Title: Gradient Centralization for Better Training Performance
+Author: [Rishit Dagli](https://github.com/Rishit-dagli)
+Date created: FILLME
+Last modified: FILLME
+Description: Implement Gradient Centralization to improve training performance of DNNs.
+"""
+"""
+# Introduction
+
+This example implements [Gradient Centralization](https://arxiv.org/abs/2004.01461), a
+new optimization technique for Deep Neural Networks by Yong et al., and demonstrates it
+on Laurence Moroney's [Horses or Humans
+Dataset](https://www.tensorflow.org/datasets/catalog/horses_or_humans). Gradient
+Centralization can both speedup training process and improve the final generalization
+performance of DNNs. It operates directly on gradients by centralizing the gradient
+vectors to have zero mean. Gradient Centralization morever improves the Lipschitzness of
+the loss function and its gradient so that the training process becomes more efficient
+and stable.
+
+This example requires TensorFlow 2.2 or higher as well as `tensorflow_datasets` which can
+be installed with this command:
+
+```
+pip install tensorflow-datasets
+```
+
+We will be implementing Gradient Centralization in this example but you could also use
+this very easily with a package I built,
+[gradient-centralization-tf](https://github.com/Rishit-dagli/Gradient-Centralization-TensorFlow).
+[gradient-centralization-tf](https://github.com/Rishit-dagli/Gradient-Centralization-TensorFlow).
+"""
+
+"""
+# Setup
+"""
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from tensorflow.keras import layers
+from tensorflow.keras.layers import Conv2D
+from tensorflow.keras.layers import MaxPooling2D
+from tensorflow.keras.layers import Dropout
+from tensorflow.keras.layers import Dense
+from tensorflow.keras.optimizers import RMSprop
+import keras.backend as K
```

Comment:
You don't actually need this module, just us `tf.something` when you need a math op.

Reply:
Yep, after removing gradient clipping we only have 1 math op for which we now use `tf.reduce_mean()`